server/channels/api4/channel.go

```diff
@@ -169,6 +169,10 @@ func updateChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 			c.Err = model.NewAppError("updateChannel", "api.channel.patch_update_channel.forbidden.app_error", nil, "", http.StatusForbidden)
 			return
 		}
+		if channel.Name != "" || channel.DisplayName != "" || channel.Purpose != "" {
+			c.Err = model.NewAppError("updateChannel", "api.channel.update_channel.cant_update_direct_or_group_messages.app_error", nil, "", http.StatusBadRequest)
+			return
+		}
```

Comment:
This is wrong and assumes that the user should only set those fields which are meant to be changed. That's what patchChannel is for. In `UpdateChannel`, users can set the old fields as unchanged as well.

We need to compare the new channel value with the old channel one, like in L187 and then throw an error accordingly. And we also need to update the `patchChannel` endpoint.

Reply:
That seems to be my misunderstanding laying out the description of the ticket. @kyeongsoosoo do you mind updating the PR according to these comments? Thanks!

Reply:
Thanks for the review. 
I've updated the PR based on your comments.